src/charm.py

```diff
@@ -632,6 +644,20 @@ def _on_mysql_pebble_ready(self, event) -> None:
             # Data directory is already initialised, skip configuration
             logger.debug("Data directory is already initialised, skipping configuration")
             self._reconcile_pebble_layer(container)
+            if self.is_new_unit:
+                # when unit is new and has data, it means the app is scaling out
+                # from zero units
+                logger.debug("Scaling out from zero units")
+                if self.unit.is_leader():
+                    # create the cluster due it being dissolved on scale-down
+                    self.create_cluster()
```

Comment:
IMHO, charm must be blocked if foreign disk attached by mistake (to avoid data damage).

Today it is not possible for K8s, but Pedro is working with Juju on it:
```
> juju scale-application mysql 0
mysql scaled to 0 units

> juju add-storage mysql/0 database=foreigndisk
ERROR Juju command "add-storage" not supported on container models    # will be supported on K8s like on VM

> juju scale-application mysql 3
...
```



----

Another reason to write to disk:
```
> juju scale-application mysql 0

# juju controller crashed... restored ...

> juju scale-application mysql 1  # did we loose units-added-to-cluster ?
```